### DISCUSSION TITLE
How to use with Firestore and realtime listening

### DISCUSSION THREAD

**Discussion by beardsleym** [BetterTyped/hyper-fetch#72] [Q&A]
Can anyone help me with a simple example for how to listen to items in a collection called books.

```typescript
import { Socket, SocketAdapterInstance } from "@hyper-fetch/sockets";
import { firebaseSocketsAdapter } from "@hyper-fetch/firebase";
import { firebaseDb } from "@/lib/firebase";

const socket = new Socket({
  url: "books/",
  adapter: firebaseSocketsAdapter(
    firebaseDb
  ) as unknown as SocketAdapterInstance,
});

export const onSnapshotReq = socket.createListener<any[]>()({
  endpoint: "",
});

export const unmount = onSnapshotReq.listen({
  callback: ({ data, extra }) => {
    console.log("data", data);
  },
});

```

a) I have a type error with the adapter, so perhaps I've imported something wrongly

> Type 'FirestoreSocketAdapterType' is not assignable to type 'SocketAdapterInstance'.
>   Types of parameters 'socket' and 'socket' are incompatible.
>     Type 'Socket<SocketAdapterType<any, any, any, any>>' is not assignable to type 'Socket<SocketAdapterType<never, FirestoreOnSnapshotExtra, { groupByChangeType?: boolean | undefined; } & FirestoreQueryParams, never>>'.
>       Types of property 'options' are incompatible.
>         Type 'SocketOptionsType<SocketAdapterType<any, any, any, any>>' is not assignable to type 'SocketOptionsType<SocketAdapterType<never, FirestoreOnSnapshotExtra, { groupByChangeType?: boolean | undefined; } & FirestoreQueryParams, never>>'.
>           Type 'SocketAdapterType<any, any, any, any>' is not assignable to type 'SocketAdapterType<never, FirestoreOnSnapshotExtra, { groupByChangeType?: boolean | undefined; } & FirestoreQueryParams, never>'.

b) I'm just not getting anything, so perhaps the enpoints, or my functions are incorrect.

**Comment by prc5** [comment]
Hey! This could be solved now with: https://github.com/BetterTyped/hyper-fetch/issues/71
Let me know if you need more assistance 👀
